packages/amplify-category-auth/src/provider-utils/awscloudformation/import/index.ts

```diff
@@ -1131,7 +1130,7 @@ const headlessImport = async (
   const projectConfig = context.amplify.getProjectConfig();
 
   // If region mismatch, signal prompt for new arguments, only in interactive mode, headless does not matter
-  if (resourceParameters.region !== Region) {
+  if (resourceParameters.region && resourceParameters.region !== Region) {
```

Comment:
Why do we need this modification? if region is undefined then the equality will not true either.

Reply:
I had to put this since `resourceParameters.region` was set to `undefined` in headless mode and the `&&` will short-circuit since`resourceParameters.region` will return `false` and the equality won't be compared.